account_check_date/views/account_payment_views.xml

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="view_account_payment_form_inherit" model="ir.ui.view">
+        <field name="name">account.payment.form.inherit</field>
+        <field name="model">account.payment</field>
+        <field
+            name="inherit_id"
+            ref="account_check_printing.view_account_payment_form_inherited"
+        />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='check_number']" position="before">
+                <field
+                    name="check_date"
+                    attrs="{'invisible': ['|', ('payment_method_code', '!=', 'check_printing'), ('check_number', '=', False)]}"
```

Comment:
```suggestion
                    attrs="{'invisible': [('payment_method_code', '!=', 'check_printing')]}"
```

Reply:
I think check_date shouldn't be invisible when check_number = False.
In case user create Payment directly, now, where is no way to enter check_date during state draft.

![Selection_164](https://user-images.githubusercontent.com/1973598/133933975-318e185a-1ef6-424c-aa7d-65cdc1020141.png)